zipline/src/commonMain/kotlin/app/cash/zipline/Handle.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline
+
+import app.cash.zipline.internal.bridge.Endpoint
+import app.cash.zipline.internal.bridge.InboundBridge
+import app.cash.zipline.internal.bridge.InboundCallHandler
+import app.cash.zipline.internal.bridge.OutboundBridge
+import kotlinx.serialization.modules.SerializersModule
+import okio.Closeable
+
+abstract class Handle<T : Any> internal constructor() : Closeable {
+  fun get(serializersModule: SerializersModule): T {
+    error("unexpected call to Handle.get: is the Zipline plugin configured?")
+  }
+
+  @PublishedApi
+  internal abstract fun get(outboundBridge: OutboundBridge<T>): T
+}
+
+fun <T : Any> Handle(serializersModule: SerializersModule, service: T): Handle<T> {
+  error("unexpected call to Handle(): is the Zipline plugin configured?")
+}
+
+// Plugin-rewritten code calls the constructor of this class.
+@PublishedApi
+internal class InboundHandle<T : Any>(
+  private val inboundBridge: InboundBridge<T>
+): Handle<T>() {
+  private var name: String? = null
+  private var endpoint: Endpoint? = null
+
+  internal fun connect(endpoint: Endpoint, name: String): InboundCallHandler {
+    check(this.endpoint == null && this.name == null) { "already connected" }
+    this.name = name
+    this.endpoint = endpoint
+
+    val serializersModule = SerializersModule {
+      include(endpoint.builtInSerializersModule)
+      include(inboundBridge.serializersModule)
+    }
+
+    val context = InboundBridge.Context(serializersModule)
+    val result = inboundBridge.create(context)
+    endpoint.inboundHandlers[name] = result
+    return result
+  }
+
+  override fun get(outboundBridge: OutboundBridge<T>): T {
+    // TODO(jwilson): change generated subtypes to populate `T` on the super call.
```

Comment:
This is an important ergonomics thing. . . if you’re writing a test you’ll have an `InboundHandle` and will want to call `get()` on it.